### DISCUSSION TITLE
SB5.1 - OCI - What will be report service looks like in Sunbird 5.1.0

### DISCUSSION THREAD

**Discussion by heungheung** [Sunbird-Ed/Community#331] [Installation]
In the draft 5.1.0 doc
  https://ed.sunbird.org/use/installing-ed-platform/release-5.1.0-draft/build
Report Service will be required to build and deploy
  Build/Core/Report                  release-4.10.0_RC6
  Deploy/Kubernetes/Report   release-5.0.0_RC2  (sunbird-devops)

however, looking into code of report service
  https://github.com/project-sunbird/sunbird-report-service/tree/release-4.10.0_RC6/helpers
is still ties with Azure Blob storage

In other branch/release tags, e.g. 5.0.0_RC2 
  https://github.com/project-sunbird/sunbird-report-service/tree/release-5.0.0_RC2/helpers
  https://github.com/project-sunbird/sunbird-report-service/blob/release-5.0.0_RC2/helpers/envHelpers.js  
it is the same.

This discussion thread had mentioned something similar
   https://github.com/Sunbird-Ed/Community/discussions/283
although the thread is marked as answered, there is no detail info. 

What will report service looks like for 5.1.0 with CSP other than Azure? or this service will no longer be used in 5.1.0?

**Comment by heungheung** [comment]
any updates?

I've gone thru the code but couldn't figure out how does the storage being used. Please help

**Reply by beepdot** [reply]
@rajeevsathish Please can you provide your suggestions here?

**Reply by Krishnaj20** [reply]
@rajeevsathish @vinukumar-vs - Please look into this thread and respond.

**Reply by vinukumar-vs** [reply]
@yravinderkumar33 Can you help here?

Is the same AZURE object will work for other CSP's? Is the naming convention has to change instead of AZURE to CSP?
https://github.com/project-sunbird/sunbird-report-service/blob/release-5.0.0/helpers/envHelpers.js#L43

**Reply by Krishnaj20** [reply]
@yravinderkumar33 - can you respond to this query?

@maheshkumargangula @vinukumar-vs

**Reply by yravinderkumar33** [reply]
Same object won't work for other CSP's. That implementation is specific to Azure. Portal team has done some changes to handle this at the Portal backend layer. Same has to replicated here as well.

cc : @rajeevsathish @rajeshkumaravel

**Reply by Krishnaj20** [reply]
@rajeevsathish - Please check and respond to this thread.

@chitranshu-keshav

**Reply by rajeshkumaravel** [reply]
@heungheung 
Here are the references for implementation of CSP in Sunbird Ed Portal 
https://github.com/Sunbird-Ed/SunbirdEd-portal/blob/release-5.1.0/src/app/helpers/cloudStorage/index.js
https://github.com/Sunbird-Ed/SunbirdEd-portal/blob/release-5.1.0/src/app/routes/reportRoutes.js#L10
https://github.com/Sunbird-Ed/SunbirdEd-portal/blob/release-5.1.0/src/app/routes/reportRoutes.js#L137

Client Cloud Services - https://github.com/Sunbird-Ed/client-cloud-services

CC: @rajeevsathish @vinukumar-vs @Krishnaj20 @yravinderkumar33 @chitranshu-keshav

**Reply by heungheung** [reply]
@rajeshkumaravel 
I am good with player. I build my local client-cloud-service and inject it to player instead of using the npm module.

my question on this thread is about report service.
